docs/proposals/scheduling/average-replicas-assign/average-replicas-assign.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Karmada distributes the replicas evenly based on spread constraint
+authors:
+- ipsum-0320
+reviewers:
+- whitewindmills
+approvers:
+- whitewindmills
+creation-date: 2024-06-28
+
+---
+
+# Karmada distributes the replicas evenly based on spread constraint
+
+## Summary
+
+In Karmada's current replica scheduling policy, we have supported four replica allocation methods: replication, static weight, aggregation based on available replicas, and dynamic weight based on available replicas.
+
+<!--
+在 Karmada 当前的调度器副本数分配策略中，我们已经支持复制、静态权重、基于可用副本数的聚合、基于可用副本数的动态权重这四种副本分配方式。
+-->
+
+However, for more complex scheduling scenarios, such as ensuring that the workload replicas are evenly distributed across selected clusters after applying cross-AZ distribution constraints, Karmada currently does not support these scenarios well.
+
+<!--
+但是对于更为复杂的调度场景，例如，在跨 AZ 分布约束之后，使工作负载的副本数尽可能地平均分配在所选择的集群中。目前 Karmada 还无法很好地支持这类场景。
+-->
+
+Therefore, we plan to introduce a new replica allocation strategy called AverageReplicas for Karmada's scheduler. This strategy will support evenly distributing the target replicas among the selected clusters, and it will:
+
+1. Adhere to spread constraints.
+2. Consider the available resources in the working clusters.
+
+<!--
+因此我们计划为 Karmada 的调度器引入一种新的副本分配策略 AverageReplicas，该策略能够支持在选定的集群中均匀分配目标副本，并且该策略将：
+1.遵守传播约束。
+2.考虑工作集群中的可用资源。
+-->
+
+## Motivation
+
+By adding the AverageReplicas replica allocation strategy, Karmada can distribute the target replicas as evenly as possible across the working clusters. At the same time, this even distribution will adhere to spread constraints and consider the available resources in the working clusters.
+
+<!--
+通过添加 AverageReplicas 副本分配策略，Karmada 能够尽可能平均地将目标副本分配到工作集群中。同时，这种平均是遵守传播约束（spread constraint）且考虑工作集群中的可用资源的。
+-->
+
+Such an allocation strategy can significantly improve the system's disaster recovery capability. Even if one cluster fails, it will not have a major impact on the overall system. This is highly beneficial for ensuring high availability of the service.
+
+<!--
+这样的分配策略能够显著提高系统的容灾能力，即使某个集群挂掉，也不会对系统整体产生较大的影响。这对于保障服务高可用是非常有益的。
+-->
+
+### Goals
+
+By adding the AverageReplicas replica allocation strategy, we ensure that replicas can be evenly distributed among the selected clusters as much as possible. This even distribution means:
+
+1. The allocation results adhere to the spread constraint (Selection phase).
```

Comment:
I guess the point is to make sure that it's as evenly distributed as possible, not spread constraints.

Reply:
> I guess the point is to make sure that it's as evenly distributed as possible, not spread constraints.
@whitewindmills yeah, I only mentioned spread constraints in passing... I will remove this part.